### PR TITLE
Storyshots Setup

### PR DESCRIPTION
**Pull request by noahub**
This PR sets up snapshot testing inside of Constellation:
- Adds Storyshots addon -- snapshots generated in the `__snapshots__` dir.
- Adds `test` to run snapshot tests and `test:update` to update snapshots.
- Sets up a `test:tsc` command to check TS types, and updates Github actions to run on push
- Moves the `defaultValue` property into the `args` object in component stories.
Screen Shot 2022-08-02 at 4 27 43 PM
Screen Shot 2022-08-02 at 4 27 09 PM

### CODE REVIEW

File: packages/constellation/fileTransformer.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// eslint-disable-next-line
+const path = require('path')
+
+module.exports = {
+  process(src, filename) {
+    return `module.exports = ${JSON.stringify(path.basename(filename))};`
```
Comment (blockdylanb) [inline packages/constellation/fileTransformer.ts:9]: what's the file all about?

File: packages/constellation/jest.config.ts [hunk 2]
```diff
@@ -128,13 +130,13 @@ export default {
...
   // The paths to modules that run some code to configure or set up the testing environment before each test
-  // setupFiles: [],
+  setupFiles: ['./jest.setup.ts'],
 
   // A list of paths to modules that run some code to configure or set up the testing framework before each test
   setupFilesAfterEnv: [
     './src/setupTests.ts',
-    './mocks/localStorage.js',
-    './mocks/dispatchEvent.js',
+    // './mocks/localStorage.js',
```
Comment (blockdylanb) [inline packages/constellation/jest.config.ts]: need these commented or can we just delete these two lines?

File: packages/constellation/fileTransformer.ts (hunk 1 above)
Comment (noahub) [reply packages/constellation/fileTransformer.ts:9]: I had to go beyond stubbing out the svg files for Jest/Storyshots to correctly resolve/consume icons and components using icons.

This resolved errors like the following when running the whole suite of snapshot tests:
Screen Shot 2022-08-02 at 2 39 57 PM

Comment (noahub) [reply packages/constellation/fileTransformer.ts:9]: @blockdylanb let me know if that makes sense. Also happy to leave a comment in this file to describe why it is needed.

File: packages/constellation/jest.config.ts (hunk 2 above)
Comment (noahub) [reply packages/constellation/jest.config.ts]: Oh nice catch, will remove these.

**Review by awiermanncasas-bc — DISMISSED** [review]

**Review by ethanjc — DISMISSED** [review]

File: packages/constellation/fileTransformer.ts (hunk 1 above)
Comment (blockdylanb) [reply packages/constellation/fileTransformer.ts:9]: @noahub i see, if you could leave a comment that would be great, thanks!

**Review by blockdylanb — APPROVED** [review]

**Pull request merged**